test/e2e_node/container_lifecycle_test.go

```diff
@@ -2415,6 +2415,39 @@ var _ = SIGDescribe(nodefeature.SidecarContainers, "Containers Lifecycle", func(
 			ginkgo.It("should start a regular container", func() {
 				framework.ExpectNoError(results.HasRestarted(regular1))
 			})
+
+			ginkgo.It("should not restart when updated with a new image", func() {
```

Comment:
is there the same test for a regular container?

Reply:
I could not find the same test for a regular container using grep. Would you prefer to see it added in this PR, or as a separate PR?

Reply:
If both behave the same - then one PR for both is OK. If we see the difference in behavior, I would suggest to send regular containers test separately and follow it with the fix for the product code + test for sidecars

Reply:
Based on [this comment](https://github.com/kubernetes/kubernetes/pull/126525#discussion_r1715679548), shouldn't we split the test cases for Regular containers into a separate PR and focus on fixing the behavior in this PR so that restartable init containers restart when their image is updated, and then add the corresponding test cases?